### ISSUE TITLE
remote_storage: update S3 SDK BehaviorVersion

### ISSUE THREAD

**Issue by problame** [neondatabase/neon#7665]
PR https://github.com/neondatabase/neon/pull/7664 upgraded AWS SDKs but didn't dare to change the `BehaviorVersion`.

To be done in the week after releasing the #7664

**Comment by skyzh** [comment]
just realized that we didn't update the behavior version for all storage services (except storage scrubber), should leave the issue open

**Comment by arpad-m** [comment]
Yeah, that's why I've reopened it. The issue was more `remote_storage` in fact instead of `storage-scrubber`
